src/sprites.rs

```diff
@@ -41,19 +42,30 @@ pub fn get_sprite(
         .replace(['.', '\'', ':'], "")
         .to_lowercase();
 
-    let path = &format!(
-        "pokesprite/pokemon-gen8/{}/{}{filename}.png",
-        if shiny { "shiny" } else { "regular" },
-        if female && !is_random { "female/" } else { "" } // Random pokemon also shouldn't follow the female rule.
-    );
-
-    Data::get(path)
-        .unwrap_or_else(|| {
-            eprintln!("pokemon not found");
-            exit(1);
-        })
-        .data
-        .into_owned()
+    // Try gen7x first then fall back to gen8
+    for gen in IntoIterator::into_iter(["gen7x", "gen8"]) {
+
+        // Skip gen7x folder if checking gen7 is disabled
+        if !gen7 && gen == "gen7x" {
+            continue
+        }
+
+        let path = &format!(
+            "pokesprite/pokemon-{}/{}/{}{filename}.png",
```

Comment:
Maybe add the gen7 implementation here instead in the `format!` call.

Reply:
I am not too sure I get what you mean?

I could do  "gen7x" if gen7 else "gen8" in the format statement and remove the for loop, that would mean there is no fallback to generation 8's sprites

Reply:
Unless you mean:
```rust

    // I hate Mr. Mime and Farfetch'd.
    filename = filename
        .replace([' ', '_'], "-")
        .replace(['.', '\'', ':'], "")
        .to_lowercase();

    // Try gen7x first then fall back to gen8
    for gen in IntoIterator::into_iter(["gen7x", "gen8"]) {

        // Skip gen7x folder if checking gen7 is disabled
        if !gen7 && gen == "gen7x" {
            continue
        }

        let path = &format!(
            "pokesprite/pokemon-{}/{}/{}{filename}.png",
            gen,
            if shiny { "shiny" } else { "regular" },
            if female && !is_random { "female/" } else { "" } // Random pokemon also shouldn't follow the female rule.
        );

        if !Data::get(path).is_some() {
            continue;
        }
    }

    Data::get(path)
        .unwrap_or_else(|| {
            eprintln!("pokemon not found");
            exit(1);
        })
        .data
        .into_owned()
}
```

Reply:
```
	let mut path: String = "".to_string();
	
    // Try gen7x first then fall back to gen8
    for gen in IntoIterator::into_iter(["gen7x", "gen8"]) {

        // Skip gen7x folder if checking gen7 is disabled
        if !gen7 && gen == "gen7x" {
            continue;
        }

        let m_path = &format!(
            "pokesprite/pokemon-{}/{}/{}{filename}.png",
            gen,
            if shiny { "shiny" } else { "regular" },
            if female && !is_random { "female/" } else { "" } // Random pokemon also shouldn't follow the female rule.
        );
		
        if Data::get(m_path).is_some() {
        	path = m_path.to_owned();
            break;
        }
    }

    Data::get(path.as_str())
        .unwrap_or_else(|| {
            eprintln!("pokemon not found");
            exit(1);
        })
        .data
        .into_owned()
```

Unless you meant something like that; apologies... my rust is not great :sweat:

Reply:
I don't think fallback is really necessary, since no other flags have fallback.

Reply:
Would that not break random? A separate list of older supported pokemon in the gen7x folder would need to be added